### ISSUE TITLE
`parameter_assignments` false negative for anonymous functions

### ISSUE THREAD

**Issue by Giuspepe** [dart-lang/linter#3237]
**Describe the issue**
The lint `parameter_assignments` does not work with anonymous functions.

**To Reproduce**
```dart
void main(){
    // `parameter_assignments` works fine with named functions
    void foo(int i) {
      i = 42; // LINT: `parameter_assignments`
      print(i);
    }
    foo(0);

    // `parameter_assignments` does not work with anonymous functions
    (int i) {
      i = 42; // false negative: should yield lint but doesn't!
      print(i);
    }(0);
}
```
**Expected behavior**
The lint `parameter_assignments` should work with anonymous functions as well.

**Comment by spkersten** [comment]
There's also no warning for constructors:
```
class Foo {
  Foo(int x) {
    x = 4;
  }
}
```
